homeassistant/components/synology_dsm/sensor.py

```diff
@@ -0,0 +1,228 @@
+    unit = hass.config.units.temperature_unit
+    use_ssl = entry.data[CONF_SSL]
+    api_version = entry.data.get(CONF_API_VERSION, DEFAULT_DSM_VERSION)
+
+    api = SynoApi(hass, host, port, username, password, unit, use_ssl, api_version)
+
+    await hass.async_add_executor_job(api.update)
+
+    sensors = [
+        SynoNasUtilSensor(api, name, sensor_type, UTILISATION_SENSORS[sensor_type])
+        for sensor_type in UTILISATION_SENSORS
+    ]
+
+    # Handle all volumes
+    if api.storage.volumes is not None:
+        for volume in entry.data.get(CONF_VOLUMES, api.storage.volumes):
+            sensors += [
+                SynoNasStorageSensor(
+                    api, name, sensor_type, STORAGE_VOL_SENSORS[sensor_type], volume
+                )
+                for sensor_type in STORAGE_VOL_SENSORS
+            ]
+
+    # Handle all disks
+    if api.storage.disks is not None:
+        for disk in entry.data.get(CONF_DISKS, api.storage.disks):
+            sensors += [
+                SynoNasStorageSensor(
+                    api, name, sensor_type, STORAGE_DISK_SENSORS[sensor_type], disk
+                )
+                for sensor_type in STORAGE_DISK_SENSORS
+            ]
+
+    async_track_time_interval(hass, api.update, SCAN_INTERVAL)
+
+    async_add_entities(sensors, True)
+
+
+class SynoApi:
+    """Class to interface with Synology DSM API."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        host: str,
+        port: int,
+        username: str,
+        password: str,
+        temp_unit: str,
+        use_ssl: bool,
+        api_version: int,
+    ):
+        """Initialize the API wrapper class."""
+        self._hass = hass
+        self.temp_unit = temp_unit
+
+        self._api = SynologyDSM(
+            host, port, username, password, use_ssl, dsm_version=api_version
+        )
+
+        self.utilisation = self._api.utilisation
+        self.storage = self._api.storage
+
+    def update(self, now=None):
+        """Update function for updating API information."""
+        self._api.update()
+        dispatcher_send(self._hass, SERVICE_UPDATE)
+
+
+class SynoNasSensor(Entity):
+    """Representation of a Synology NAS Sensor."""
+
+    def __init__(
+        self,
+        api: SynoApi,
+        name: str,
+        sensor_type: str,
+        sensor_info: Dict[str, str],
+        monitored_device: str = None,
+    ):
+        """Initialize the sensor."""
+        self.sensor_type = sensor_type
+        self._name = f"{name} {sensor_info[0]}"
+        self._unit = sensor_info[1]
+        self._icon = sensor_info[2]
+        self.monitored_device = monitored_device
+        self._api = api
+
+        if self.monitored_device is not None:
+            self._name = f"{self._name} ({self.monitored_device})"
+
+        self._unsub_dispatcher = None
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return self._name
```

Comment:
Is there a better unique id here?

https://developers.home-assistant.io/docs/entity_registry_index/

Reply:
Sure, will use same unique_id mechanism as config_flow